x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/grouping/GroupingFunction.java

```diff
@@ -45,4 +41,20 @@ public BiConsumer<LogicalPlan, Failures> postAnalysisPlanVerification() {
         };
     }
 
+    public abstract static non-sealed class NonEvaluatableGroupingFunction extends GroupingFunction {
```

Comment:
Should we add some javadocs here? Maybe mentioning that "it's (currently?) used for stateful grouping functions, which make them effectively non-evaluatable before the agg, so its value can't be used inside the aggs".

Even if it has some high level examples, I think it will be useful for whoever has to reuse them in the future

Reply:
Yes, added some docs.